unreleased/multi-package.md

```diff
@@ -0,0 +1,462 @@
+
+The `main` package depends on `my-lib`, which itself depends on `my-lib-helper`.
+
+```
+> tree
+.
+├── multi-package.yaml
+├── libs
+│   ├── multi-package.yaml
+│   ├── my-lib
+│   │   ├── daml
+│   │   │   └── MyLib.daml
+│   │   └── daml.yaml
+│   └── my-lib-helper
+│       ├── daml
+│       │   └── MyLibHelper.daml
+│       └── daml.yaml
+└── main
+    ├── daml
+    │   └── Main.daml
+    └── daml.yaml
+```
+
+Looking inside the most relevant files:
+
+- `libs/multi-package.yaml`
+  ```yaml
+  packages:
+  - ./my-lib
+  - ./my-lib-helper
+  ```
+- `multi-package.yaml`
+  ```yaml
+  packages:
+  - ./main
+  projects:
+  - ./libs
+  ```
+- `main/daml.yaml`
+  ```yaml
+  version: 1.0.0
+  ...
+  data-dependencies:
+  - ../libs/my-lib/.daml/dist/my-lib-1.0.0.dar # main depends on my-lib
+  ```
+
+As before, we can run `daml build --all` from the root of the project to build all libraries and `main`.
+
+```bash=
+> # From the root of the project:
+> daml build --all
+Building "my-lib-helper"...
+Building "my-lib"...
+Building "main"...
+```
+
+However, if we run `daml build --all` from the `libs` directory, it'll traverse the directory tree and first find the `libs/multi-package.yaml`. Because `libs/multi-package.yaml` only refers to `my-lib` and `my-lib-helper`, only those will be built.
+
+```bash=
+> cd libs/
+> daml build --all
+Building "my-lib-helper"...
+Building "my-lib"...
+# Main is *not* built, because libs/multi-package.yaml was used
+```
+
+You can explicitly specify the use of the outer `multi-package.yaml` from within `libs` using `--multi-package-path ../multi-package.yaml`.
+
+```bash=
+> cd libs/
+> daml build --all --multi-package-path ../multi-package.yaml
+Building "my-lib-helper"...
+Building "my-lib"...
+Building "main"... # Main *is* built, because the root multi-package.yaml was used
+```
+
+## Migrating from Previous Workflows
+
+Because multi-package builds are a purely opt-in additive behaviour over existing `daml build`, if you don't plan to use multi-package builds you do not need to change anything about your package configurations.
+
+If you do plan to use multi-package files, your `daml.yaml` files will not need updating in general because the `multi-package.yaml` needs only a path to a folder with a `daml.yaml`.
+
+There is one caveat: `multi-package.yaml` assumes that all dependent packages refer to DAR files by the locations where `daml build` would normally put them. This means that if you have shell scripts that move a package's DAR file after build, you will need use the `--output` flag instead, so that `multi-package.yaml` can statically find the DAR's expected output location.
+
+### The `--output` flag
+
+The `--output` flag can be used to specify the location of the DAR generated by `daml build`. This is not new, however it is important to note that the multi package feature extends the use-cases of this flag.
```

Comment:
```suggestion
The `--output` flag can be used to specify the location of the DAR generated by `daml build`. This is not new, however it is important to note that the multi-package feature extends the use cases of this flag.
```

Reply:
I didn't mark all hyphenation issues or punctuation issues generally -- will do that in the Daml docs repo